opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -227,29 +228,59 @@ public void setNestingMaximum(int nestingMaximum) {
         syncWriteConfiguration(nestingMaximum, Configuration::setNestingMaximum);
     }
 
-    public int getCommandTimeout() {
-        return syncReadConfiguration(Configuration::getCommandTimeout);
+    public int getCommandTimeout(CommandTimeoutType cmdType) {
+        switch (cmdType) {
+            case INDEXER:
+                return getIndexerCommandTimeout();
+            case INTERACTIVE:
+                return getInteractiveCommandTimeout();
+            case WEBAPP_START:
+                return getWebappStartCommandTimeout();
+            case RESTFUL:
+                return getRestfulCommandTimeout();
+        }
+
+        throw new InvalidParameterException("invalid command timeout type");
```

Comment:
Would not `IllegalArgumentException` make more sense? `InvalidParameterException` is from `security` package and is not that commonly used.

Reply:
sounds reasonable, changed.